crypto/defaults.c

```diff
@@ -60,44 +60,46 @@ static char *modulesdirptr = NULL;
  *
  * @return A pointer to a char array containing the registry directories.
  */
-static char *get_windows_regdirs(char *dst, LPCTSTR valuename)
+static char *get_windows_regdirs(char *dst, DWORD dstsizebytes, LPCWSTR valuename)
 {
     char *retval = NULL;
 # ifdef REGISTRY_KEY
-    DWORD keysize;
+    DWORD keysizebytes;
     DWORD ktype;
     HKEY hkey;
     LSTATUS ret;
     DWORD index = 0;
-    LPCTCH tempstr = NULL;
-   
+    LPCWSTR tempstr = NULL;
+
     ret = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
                        TEXT(REGISTRY_KEY), KEY_WOW64_32KEY,
                        KEY_QUERY_VALUE, &hkey);
     if (ret != ERROR_SUCCESS)
         goto out;
 
-    ret = RegQueryValueEx(hkey, valuename, NULL, &ktype, NULL,
-                          &keysize);
+    // Always use wide call so we can avoid extra encoding conversions on the output
+    ret = RegQueryValueExW(hkey, valuename, NULL, &ktype, NULL,
+                          &keysizebytes);
```

Comment:
off-by-one indentation error here. Please line up `&` with `h` on the previous line